engine/src/main/java/com/datatorrent/stram/client/StramAgent.java

```diff
@@ -421,7 +421,11 @@ private StramWebServicesInfo retrieveWebServicesInfo(String appId)
         permissionsInfo = new JSONObject(IOUtils.toString(is));
       } catch (IOException ex) {
         // ignore if file is not found
-        LOG.info("Exception in accessing permissions.json", ex);
+        if (LOG.isDebugEnabled()) {
```

Comment:
Since the feature is rarely used, checking for file existence would be equivalent to checking whether this feature is enabled or not and it seems like a better approach. If the file does exist, the file not found exception would be handled generally with the IOException.

However, if the situation were reversed and this feature was indeed used regularly, then we would directly handle file not found exception as the exceptional scenario and fallback to desired fallback behavior.

Since this feature does exist today, should we even be favoring a particular usage (or in this case disusage) pattern? Should we just treat it in good faith and assume that it will be used and deal with exceptional scenarios. We could have a separate discussion on whether this feature is still needed or other alternatives if it is not being used.